src/seer/automation/codegen/prompts.py

```diff
@@ -212,3 +214,104 @@ def format_pr_review_plan_step(diff_str: str):
         ).format(
             diff_str=diff_str,
         )
+
+
+class _RelevantWarningsPromptPrefix:
+    """
+    Stores common prompt prefixes to maximize prompt cache hits (for OpenAI calls).
+    """
+
+    @staticmethod
+    def format_system_msg():
+        # https://github.com/codecov/bug-prediction-research/blob/f79fc1e7c86f7523698993a92ee6557df8f9bbd1/src/scripts/ask_oracle.py#L79-L81
+        return textwrap.dedent(
+            """\
+            You are a senior developer with extensive knowledge about code and static analysis warnings.
+            You always consider carefully the context of the code and the error before making a decision.
+            You are tasked with analyzing a single issue and providing a detailed explanation of the issue and the context in which it occurs.
+            """
+        )
+
+    @staticmethod
+    def format_prompt_error(formatted_error: str):
+        """
+        The error makes up the bulk of the prompt b/c of the stacktrace.
+        Put it right after the system message so that the system message and error are cached
+        together.
+        """
+        return textwrap.dedent(
+            f"""\
+            Here is an issue we had in our codebase:
+
+            {formatted_error}
+            """
+        )
```

Comment:
```suggestion
        return textwrap.dedent(
            """\
            Here is an issue we had in our codebase:

            {formatted_error}
            """
        ).format(formatted_error=formatted_error)
```
dedent won't correctly work if the formatted error has indents in it too